velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -379,5 +379,38 @@ TEST_F(DateTimeFunctionsTest, dayofWeekTs) {
   }
 }
 
+TEST_F(DateTimeFunctionsTest, dateDiffDate) {
+  const auto dateDiff = [&](std::optional<int32_t> endDate,
+                            std::optional<int32_t> startDate) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_diff(c0, c1)", {endDate, startDate}, {DATE(), DATE()});
+  };
+
+  const auto dateDiffAlias = [&](std::optional<int32_t> endDate,
+                                 std::optional<int32_t> startDate) {
+    return evaluateOnce<int32_t, int32_t>(
+        "datediff(c0, c1)", {endDate, startDate}, {DATE(), DATE()});
+  };
+
+  // Simple tests.
+  EXPECT_EQ(-1, dateDiff(parseDate("2019-02-28"), parseDate("2019-03-01")));
+  EXPECT_EQ(-358, dateDiff(parseDate("2019-02-28"), parseDate("2020-02-21")));
+  EXPECT_EQ(0, dateDiff(parseDate("1994-04-20"), parseDate("1994-04-20")));
+
+  // Account for the last day of a year-month.
+  EXPECT_EQ(395, dateDiff(parseDate("2020-02-29"), parseDate("2019-01-30")));
+
+  // Check Large date.
+  EXPECT_EQ(
+      -737790, dateDiff(parseDate("2020-02-29"), parseDate("4040-02-29")));
+
+  // Negative year.
+  EXPECT_EQ(
+      2147474628,
```

Comment:
Shouldn't this be a negative value?

Reply:
Here, the value overflows. Just refined the test comment. Thanks!